enhancements/insights/up-to-date-gathering.md

```diff
@@ -0,0 +1,250 @@
+* Control the lifecycle of the gatherer from *main* insights-operator container (different executable as part of the same image): this includes using newer versions
+of the image when available, as well as fallback mechanism to older version on failures.
+
+### Non-Goals
+
+
+## Proposal
+
+Separate The Insights Operator pod in two containers the Operator *main* and 
+Insighs Operator *gatherer*.
+
+The *main* will be similar to today's operator. It would like a proper operator 
+drive *gatherer*, but keep all the non-gathering logic, like uploading, status updates etc in it.
+It will be periodically checking for new versions in a 4.x-fast-track image on quay.io and eventually update the
+Insights Operator Gatherer container to latest 4.x-fast-track image url.
+It will not contain the Gathering methods and it will use Rest to reach *gatherer*, which will be 
+responsible for collection and saving data to storagePath (/var/lib/insights-operator) and this volume will be 
+shared among two containers.
+The *main* will instead of local Gather call remote Gather as Api, but will keep all the status handling and uploading
+in *main*.
+
+The Insights Operator *gatherer* will become just rest server on operators framework exposing Gather endpoint.
+Instead of returning all the data, the method will save the data using disk flusher to storagePath and return
+only status. (This was discussed as an improvement to save memory in IO for larger datasets, like logs).
+It would have same security role as today.
+The proposal will be implemented in three phases, described in [Implementation phases](#implementation-phases).
+
+### Integrating with CVO
+
+There is still to be described how we are going to keep upgraded Operator Deployment after the image url change.
+Cluster Version Operator is reverting Operator Deployment config even if I just add a new container.
+There are two possible ways. Create separate, non CVO managed Deployment for Gatherer and make Gatherer separate Pod,
+or find some way how to make CVO keep changes in second container image after the Gatherer image update. This is still
+being investigated.
+
+### Handling errors and failback
+
+If the Insights Operator *Gatherer* runs in a fast-track (non release) image,
+*main* will be checking also health of *gatherer* container.
+Besides states it is reporting today, like when it cannot upload, it will be reporting
+error from *gatherer* call. When *host* detect an issue with *gatherer it will 
+update image back to its own image url, which will be stable release.
+*Main* would also need checking crashlooping state of *gatherer* for example if image cannot be downloaded.
+
+### Image selection
+
+There might be various ways how to select the good new image.
+The initial idea is to check for new versions check in quay.io. We would need to create new tag in quay, 
+for example fast-track, and we would be checking
+```
+quay.io/openshift/origin-insights-operator:4-5-fast-track
```

Comment:
We need to support clusters where all images are mirrored to a private registry. I didn't see anything in the enhancement that talks about how it will work in those environments. All images in the core platform installation must be referenced by SHA today to work with ImageContentSourcePolicy.